sr_port/push_lvval.c

```diff
@@ -34,12 +37,6 @@ lv_val *push_lvval(mval *arg1)
 	error_def(ERR_STACKOFLOW);
 	error_def(ERR_STACKCRIT);
 
-	/* Note that since this is only (currently) used by call-ins and no TP
-	   transaction can be (currently) be active during a call-in, we do not
-	   worry about setting up tp_var structures as is done in lv_newname().
-	   The assert below will catch if this condition changes.
-	*/
-	assert(!dollar_tlevel);
```

Comment:
Yes I suspect so. I could not find any reason why the comment was valid. If you do find a reason and/or a test case that demonstrates an issue, let me know.

Reply:
I'm not sure I have time to test it right now but I suspect if a var created thusly were specified on the save/restore list of a TSTART, the local var was modified inside the transaction, and some TPRESTARTs happened, it might cause an explosion now that TP can nest. That's when those additional structures are used - during save and restore during a transaction that gets restarted.

Reply:
Good point. I will try to come up with a test case for this.

Reply:
I tried to come up with a test but could not find one that demonstrates any code issue. I suspect that is because push_lvval() operates on mvals without an associated local variable name and those are bound to a local variable name at op_bindparm time but at that time a lv_newname() is done. So push_lvval() does not need to do anything more in my opinion. If you still think there is an issue, please make a note to come up with a test case even if later.

Reply:
It's probably orphaned base vars (due to aliasing) or something. I'll make a note to try to come up with a failing test case for it.